devenv/lib/archive.py

```diff
@@ -20,17 +21,34 @@ def atomic_replace(src: str, dest: str) -> None:
     os.replace(src, dest)
 
 
-def download(url: str, sha256: str, dest: str = "") -> str:
+def download(
+    url: str,
+    sha256: str,
+    dest: str = "",
+    retries: int = 3,
+    retry_exp: float = 2.0,
+) -> str:
+    assert retries >= 0
+
     if not dest:
         cache_root = f"{home}/.cache/sentry-devenv"
         dest = f"{cache_root}/{sha256}"
         os.makedirs(cache_root, exist_ok=True)
 
     if not os.path.exists(dest):
-        try:
-            resp = urllib.request.urlopen(url)
-        except HTTPError as e:
-            raise RuntimeError(f"Error getting {url}: {e}")
+        retry_sleep = 1.0
+        while retries >= 0:
```

Comment:
```suggestion
        while retries > 0:
```

Reply:
why? currently if retries = 0 then we'll try once and if err, raise the runtimeerror

Reply:
Fair enough